ultranest/utils.py

```diff
@@ -90,8 +90,13 @@ def make_run_dir(log_dir, run_num=None, append_run_num=True):
     _makedirs(log_dir)
 
     if run_num is None or run_num == '':
-        run_num = (sum(os.path.isdir(os.path.join(log_dir,i))
-                       for i in os.listdir(log_dir)) + 1)
+        for run_num in range(1,1000):
```

Comment:
Please add a comment before here saying 
`# find next free run_num`
or similar

Reply:
I've just added
```python
        # loop over existing folders of the form log_dir/runX                                                                                                                                                                                                 
        # to find next available run_num (up to the hardcoded maximum of 1000)                                                                                                                                                                                
```

Reply:
Please make the upper limit a parameter of the function, set the default to 10000, and mention it in the ValueError "log directory '%s' already contains the maximum number of run subdirectories (%d)".

Reply:
`max_run_num` may be a good name.

Reply:
I see that in the run_num docstring it says `prefix/1/` which should be `prefix/run1/`, could you please fix that as well?

Reply:
Done!

I also changed the test from `os.path.isdir` to `os.path.exists` because I presume we wouldn't want to overwrite a file that happens to be called something in the sequence.

Reply:
> I see that in the run_num docstring it says `prefix/1/` which should be `prefix/run1/`, could you please fix that as well?

I think I committed and pushed while you wrote this. It's gone into a subsequent commit but I guess you'll squash before merging or rebasing.